src/components/ChecFileUploader.vue

```diff
@@ -0,0 +1,152 @@
+
+Dropzone.autoDiscover = false;
+
+export default {
+  name: 'ChecFileUploader',
+  components: {
+    ChecButton,
+    FileRow,
+  },
+  props: {
+    /**
+     * The files that will be rendered as <file-row> components
+     */
+    files: {
+      type: Array,
+      default: () => [],
+    },
+    /**
+     * The endpoint to upload files to
+     */
+    endpoint: {
+      type: String,
+      default: '/',
+    },
+  },
+  data() {
+    return {
+      inProgressFiles: [],
+    };
+  },
+  mounted() {
+    this.dropzone = new Dropzone(this.$el, {
+      url: this.endpoint,
+      createImageThumbnails: false,
+      previewsContainer: false,
+      clickable: '[data-dropzone-clickable]',
+      hiddenInputContainer: this.$el,
+    });
+    const vm = this;
+    this.dropzone.on('complete', (file) => {
+      vm.inProgressFiles = [
+        ...vm.inProgressFiles,
+        file,
+      ];
+      this.emitFileUploadingComplete(file);
+    });
+    this.dropzone.on('addedfile', (file) => {
+      vm.inProgressFiles = [
+        ...vm.inProgressFiles,
+        file,
+      ];
+      this.emitFileAdded(file);
+    });
+    this.dropzone.on('uploadprogress', (file) => {
+      vm.inProgressFiles = [
+        ...vm.inProgressFiles,
+        file,
+      ];
+      this.emitFileUploading(file);
+    });
+  },
+  methods: {
+    handleRemovingFile(file) {
+      this.$emit('remove-file', file);
+    },
+    getNonReactiveFileObject(file) {
+      return {
+        upload: { ...file.upload }, name: file.name, size: file.size, type: file.type, status: file.status,
+      };
+    },
+    emitFileUploading(file) {
+      /**
+       * Emitted when the file has begun uploading, derived from Dropzone.js' uploadprogress event
+       *
+       * @event file-uploading
+       * @type {Object}
+       */
+      this.$emit('file-uploading', this.getNonReactiveFileObject(file));
+    },
+    emitFileAdded(file) {
+      /**
+       * Emitted when the file has been initially added, derived from Dropzone.js' addedfile event
+       * @event file-added
+       * @type {$event}
+       */
+      this.$emit('file-added', this.getNonReactiveFileObject(file));
+    },
+    emitFileUploadingComplete(file) {
+      /**
+       * Emitted when the file has completed uploading, derived from Dropzone.js' file-uploading-complete event
+       * @event file-uploading-complete
+       * @type {$event}
+       */
+      this.$emit('file-uploading-complete', this.getNonReactiveFileObject(file));
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+.chec-file-uploader {
+  @apply relative flex flex-col w-full h-auto flex rounded shadow-inner bg-gray-gradient flex justify-center;
+
+  &__inner {
+    @apply relative border border-dashed border-gray-400 rounded m-2 p-8;
+  }
+
+  .button {
+    @apply mx-auto;
+  }
+
+  &__input {
+    display: none;
+  }
+
+  .file-rows-container {
+    @apply w-full flex flex-col mb-2;
+
+    .chec-file-row {
+      @apply mb-2;
```

Comment:
We've been using `space-y-2` for this in other places. We currently have to apply it with a classname (not `@apply`), as Tailwind doesn't support adding that helper with `@apply` yet.